feat: install Grafana with Prometheus datasource

xcoulon opened the pull request:
- create a ServiceAccount with `cluster-monitoring-view` role
- retrieve the ServiceAccount's token and use it as a bearer token
to connect to Thanos Querier (which is itself "protected with
an openshift/oauth-proxy sidecar container)

Fixes https://issues.redhat.com/browse/CRT-809

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
feat: install Grafana with Prometheus datasource

- create a ServiceAccount with `cluster-monitoring-view` role
- retrieve the ServiceAccount's token and use it as a bearer token
to connect to Thanos Querier (which is itself "protected with
an openshift/oauth-proxy sidecar container)

Fixes https://issues.redhat.com/browse/CRT-809

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
add missing annotation to redirect upon oauth login

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
parameterized mapping method (use 'claim' to auto-provision)

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
use grafana 7.3.0 and use param for SA

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

alexeykazakov commented on config/monitoring/cluster_monitoring_config.yaml:
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: openshift-monitoring
```
It's going to be a problem in OSD.
I don't think we can't create anything in openshift-* namespaces in OSD.
So, I guess user workload monitoring is not enableable in OSD? Worth checking with app-sre and/or platform-sre folks.

xcoulon replied:
yeah, we need to check that indeed 🤯

alexeykazakov commented on config/monitoring/user_workload_monitoring_config.yaml:
```diff
@@ -0,0 +1,7 @@
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
```
Same here.

alexeykazakov commented on config/oauth/idp.yaml:
```diff
@@ -5,7 +5,7 @@ metadata:
 spec:
   identityProviders:
-  - mappingMethod: lookup
+  - mappingMethod: ${MAPPING}
```
Are you parameterizing it for local testing? We should never use anything but "lookup" in our stage and prod env.
If it's for local development then fine. But let's mention it in readme.

xcoulon replied:
yes, the default value remains `lookup` as you can see in https://github.com/codeready-toolchain/toolchain-infra/pull/34/files#diff-3c99f4ffee6524f96786b5c5a37a00a16a3a5c7ba1b44975e251578a31a3a632R14-R17
and yes, I wanted to use the `claim` mapping method for local testing. I'll mention that in the README ;)

xcoulon replied:
done in ad952ea

xcoulon committed:
Update README about the 'MAPPING' env var

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
fix auth issues

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
refactor ns setup and include dashboards

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
updated dashboard

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
update dashboard panels

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
fix queries on memory and CPU panels

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
refine legend/values on CPU graph

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

xcoulon committed:
host/member cluster setup

run something like `KUBECONFIG_HOST=~/.kube/config.host.staging KUBECONFIG_MEMBER=~/.kube/config.member.staging ./setup_monitoring.sh`
to enable UWM on both operators and install Grafana on the host cluster.

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

MatousJobanek commented:
@xcoulon I guess that we can close this PR, right?

xcoulon commented:
ah yes, indeed! Sorry for the late response 😳
